### ISSUE TITLE
Preference about grouping

### ISSUE THREAD

**Issue by johnbent** [tekinged/missing#363]
I was just looking up 'se' and I noticed there are a ton of expressions which show up as individual entries:
https://tekinged.com/?lookup=se&direction=pe
![image](https://user-images.githubusercontent.com/2047294/196818778-fafd1614-3727-4fed-ae15-527e648b28b0.png)

I think it would look cleaner and be easier for the user if these are grouped into some of the base words. Something more like, 

![image](https://user-images.githubusercontent.com/2047294/196818895-f509574f-1dbb-4f93-8297-fe8e00a7a6fd.png)

Does anyone have strong feelings that they are better left as they are now as separate individual entries?

**Comment by smith-371** [comment]
I guess this group should have merged only separate expressions with SE, because it's too frequent word.

**Comment by johnbent** [comment]
In this case, I was planning to merge them into the less-common word. For example, I was thinking to merge 'er se el mlo merek el rak' into 'rak'. The only problem is that the 'see also' would end up pointing to itself. But I can clean that up with some fancy mysql.

**Comment by ChrisPerrette** [comment]
Maybe this would be better as a grammatical explanation. There are so many common constructions with these words that it would be helpful to new learners who would not intuitively understand their meaning.

**Comment by smith-371** [comment]
> In this case, I was planning to merge them into the less-common word. For example, I was thinking to merge 'er se el mlo merek el rak' into 'rak'. The only problem is that the 'see also' would end up pointing to itself. But I can clean that up with some fancy mysql.

yep. I also did it so in the Woleaian dictionary, which doesn't work anymore(

**Comment by johnbent** [comment]
> Maybe this would be better as a grammatical explanation. There are so many common constructions with these words that it would be helpful to new learners who would not intuitively understand their meaning.

That would be nice. How specifically do you propose we do this? We need someone to write the grammatical explanation and then we need to figure out how to share that with people. Just another link from the top menus or some other method?

**Comment by ChrisPerrette** [comment]
I think under the DOSUUB tab, we could have a page called "Common Constructions" or similar. In this we could have sections labeled "SEI," "KIREL," "NGERANG," "MEKERANG," etc., and give clear examples of different ways these are used.

For example, "Se el bo kumtab a bechik e ng kmal mo dmeu a renguk." This means, "When I think about my spouse, I become very happy." But as a beginner, I would have thought this meant something like, "What I think about my wife, etc."

As another example, "Ak mesuub a tekoi er Belau el kirel me lsebechek el medenge a cheldechederir a ongalek er a bechik." This means "I study Palauan to be able to understand my spouse's family's conversations." But as a beginner, I would have thought that this meant something like, "I study the language that pertains to Palau and I can understand..."

The specific nuance is lost with new learners in many of these phrases. I have been studying Palauan on and off for about 9 years now, and I know I would have appreciated this as a beginner.

**Comment by johnbent** [comment]
Actually, they are already grouped in the root words. There seems to be some sophisticated logic (I previously wrote and forgot) that ungroups expressions in some search results. Cool!

There are some dangling expressions that we should probably group:


mysql> select pal from all_words3 where pos like 'expr.' and id=stem order by pal;
+-----------------------------+
| pal                         |
+-----------------------------+
| be betei                    |
| bkul a chim                 |
| boluaisei                   |
| chad el ngais               |
| chelius el techull          |
| chemachel era cheldebechel  |
| cheruchem a ngerel          |
| dokomademo                  |
| ediul                       |
| elecha el kesus             |
| iikemuu                     |
| imamade                     |
| kata er a keik              |
| Ke ua ngarang?              |
| kemanget a chimal           |
| mengelekel a ngikel         |
| mottai nai                  |
| ordomel er a beluu          |
| ouchaet a chiklel           |
| rubekul a ordomel           |
| teluidel a chural           |
| terkel a mlai               |
| torech er a redil           |
| tsudeni                     |
| tsuide                      |
| tututututut                 |
| uchelir a rechad er a Belau |
| ukab er a rengul            |
+-----------------------------+
28 rows in set (0.13 sec)

**Comment by ChrisPerrette** [comment]
Is there a way to custom order search results?

**Comment by johnbent** [comment]
Not right now. Just alpha which is the default. How else would you like to sort?
________________________________

**Comment by ChrisPerrette** [comment]
Well, like you said, parts of speech and root words would be nice. Also the ability to put the matching entries before the matching definitions. I can't tell you how many times I have searched a common word only to have to scroll through a hundred entries that have the word in their definitions but not the entry itself. Perhaps also the ability to prioritize main entries over sub-entries.

I am sure I am asking for a lot, but I think it would simplify a lot of searches.

**Comment by johnbent** [comment]
another page I want under dosuub is something like 'little words'. Would be great to have a beginner page explaining the words like 'a', 'el', 'le', 'er'.

**Comment by johnbent** [comment]
> Also the ability to put the matching entries before the matching definitions.

The rest might take a bit more effort. This particular request makes a ton of sense though. And this should just be the default. Adding preferences is more work. Changing the default is much easier. I think this is a great idea and I'll try to find time to do this. I have seen this same problem myself when searching for a small common word.

> Perhaps also the ability to prioritize main entries over sub-entries.

Not sure what this means. Can you provide an example search and explain what you are thinking please?

**Comment by ChrisPerrette** [comment]
Well, if you look at your SE search, you’ll notice that SE appears as part of an expression many times before it displays the main entry SE/SEI/ISEI.

This actually brings up a bug I have noticed. The search won’t actually consider any of the alternative spellings after the first. Like if you search SEI, it does not highlight the word SEI I the main entry, SE/SEI/ISEI.

**Comment by johnbent** [comment]
@ChrisPerrette , thanks for the feedback and helping make tekinged better! I'm a teeny bit confused though. What you just explained is already what I understood you to be asking for when you said, "Also the ability to put the matching entries before the matching definitions." Cool. I'm gonna work on that!

So when you also said, "Perhaps also the ability to prioritize main entries over sub-entries," I thought this was different from the above. But now I think it's just the same request in different words. Cool. I'm gonna work on that! But, if these are different requests, please help me understand the difference. 

> This actually brings up a bug I have noticed. The search won’t actually consider any of the alternative spellings after the first. Like if you search SEI, it does not highlight the word SEI I the main entry, SE/SEI/ISEI.

Also, not sure I understand this. Are you suggesting that when you search for a word, it should highlight (and search) for the word and its variants? That's an interesting idea. Right now the search results for SE are different than the search results for SEI and ISEI. Since we have them as variants, are you suggesting that the search results for all three should be the same (and should be the merged searches for all three)? That's an interesting idea.

Thanks for all the great feedback!

**Comment by ChrisPerrette** [comment]
Sorry if I was confusing. I would like the order of search results to be like this, using SE for example.
1. Main entries (like SE)
2. Expressions containing the word (like ER SE EL ME EL RAK)
3. Palauan definitions containing the word

I think this would be the most useful order for searches. Maybe the structure of tekinged is not conducive to this, but if it is, it would be awesome.

----------------------------------

Concerning the spelling alternatives, what you said sounds really cool. It would be nice to see the same word spelled differently. The bug I was referring to is not what I thought. I did some testing and I discovered that the search functionality works, but it doesn't highlight the entry. For example, if you search for CHAKKAI, an alternative of AKAI, it will show the word but not highlight it. So it is less of a bug than I thought it was.

**Comment by johnbent** [comment]
Update: this is closer to working now. When you search for a root word, then the search results are order as so:
- main entry
- expressions containing the main entry
- other entries referencing the search term

However, if you search for a branch word (e.g. uai is a variant of ua and mlil is a n.poss.s3 of mlai), then the ordering doesn't pull the main entry to the top. To solve this, we need to make it so that all searches reroute to the root word. However, we can't just change the search term because we want the highlighting to still highlight the original user-supplied search term.

Note to self: resume work at functions.php: 1364 (if line numbers have changed, search for 'exact')
